### ISSUE TITLE
copydoc on template function causes warning: @copybrief or @copydoc target not found

### ISSUE THREAD

**Issue by scott-zhong** [doxygen/doxygen#10640]
copydoc on template function causes `warning: @copybrief or @copydoc target not found`. doxygen-1.8.20 does not generate the warning but doxygen-1.9.0 and later does, include 1.11.0 (trunk).

doxygen-1.8.20:

```
doxygen -g
doxygen Doxyfile
```

doxygen-1.9.0:

```
doxygen -g
doxygen Doxyfile
...
Computing tooltip texts...
C:/dev/doxygen-1.9.0.windows.x64.bin/test.h:17: warning: @copybrief or @copydoc target 'foo(T)' not found
C:/dev/doxygen-1.9.0.windows.x64.bin/test.h:33: warning: @copybrief or @copydoc target 'foo(T, T)' not found
...
```

trunk:

```
doxygen -g
doxygen -x Doxyfile
# Difference with default Doxyfile 1.11.0 (5fa510c0901e7b679471473f88bdc50ec44a6bee*)
doxygen Doxyfile
...
Computing tooltip texts...
/mnt/c/dev/Ubuntu Latest GCC Release build artifacts/test.h:17: warning: @copybrief or @copydoc target 'foo(T)' not found
/mnt/c/dev/Ubuntu Latest GCC Release build artifacts/test.h:33: warning: @copybrief or @copydoc target 'foo(T, T)' not found
...
```

Testcase file:
[testcase.zip](https://github.com/doxygen/doxygen/files/14226974/testcase.zip)

**Comment by albert-github** [comment]
Looks like the problem is introduced in doxygen version 1.9.3 and is triggered by
```

Base<unsigned> count_;
```

The problem looks like to be introduced in:
```
Commit: e64a05126d4393c111d9fb9872a76b33371272e8 [e64a051]
Date: Thursday, December 9, 2021 8:26:39 PM

issue #8948: collaboration diagram should include smart pointer members
```
when replacing:
```
std::string s = result.mid(p).str();
```
by
```
std::string s = name.mid(p).str();
```
in the current case the old ("result") value was the empty string whilst the new ("name") string is ` unsigned >`

Probably we should also search for something like the closing `>` and remove it as well.


Note: The output of 1.9.2 and the current output look the same, but for more complicated situations this might not be the case anymore.

**Comment by doxygen** [comment]
@scott-zhong Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
